### PR TITLE
chapter10_manage_user

### PR DESCRIPTION
**Pull request by tenyiting**
chapter10_manage_user

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,82 @@
 class UsersController < ApplicationController
-  def show
+  before_action :load_user, only: :show
+  before_action :logged_in_user, only: %i(index edit update destroy)
```
Comment (ghost) [inline app/controllers/users_controller.rb]: use except please

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<% if object.errors.any? %>
+    <div id="error_explanation">
+      <div class="alert alert-danger">
+        <%= t (:the_form_contains)%>
```
Comment (ghost) [inline app/views/shared/_error_messages.html.erb]:
```suggestion
        <%= t(:the_form_contains) %>
```

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,7 +1,82 @@
...
+      store_location
+      flash[:danger] = t(:pls_log_in)
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    @user = User.find_by(id: params[:id])
+    return if current_user?(@user)
+    flash[:danger] = t(:n_authorized)
```
Comment (anlvq-1058) [inline app/controllers/users_controller.rb:75]: Are you use i18n?

**Review by ghost — COMMENTED** [review]
cmt

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (tenyiting) [reply app/controllers/users_controller.rb:75]: yes.. why?

**Comment by tenyiting** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**